cg/apps/demultiplex/sample_sheet/sample_sheet_validator.py

```diff
@@ -156,46 +156,63 @@ def _validate_override_cycles(self) -> None:
             except OverrideCyclesError as error:
                 raise SampleSheetError from error
 
+    def _validate_sample_sheet_is_correct_type(self, bcl_converter: str) -> None:
+        """Determine if the sample sheet content corresponds to the provided converter.
+        Raises:
+            SampleSheetError if the sample sheet is not of the correct type.
+        """
+        expected_type: Type[FlowCellSample] = BCL_CONVERTER_TO_FLOW_CELL_SAMPLE[bcl_converter]
+        if get_sample_type_from_content(self.content) is not expected_type:
+            message: str = f"Sample sheet is not {bcl_converter}"
+            LOG.error(message)
```

Comment:
I found this useful for the testing and also for production, so that we know when a BCL2FASTQ sample sheet is regenerated. I think it does not hurt that much

Reply:
Sure, keep it!